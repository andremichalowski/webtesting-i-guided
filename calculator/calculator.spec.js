const { add } = require('./calculator.js');

describe('learning about tests, assertions and matchers', () => {
  it('sanity check', () => { // the test
    expect(2 + 3).toBe(5) // an assertion
    expect(2 + 2).toBe(4) // another assertion
  })

  test('more sanity check', () => {
    expect(0).not.toBeTruthy()
    expect([]).toBeTruthy()
    expect(5).toBeLessThan(6)
    expect(5).toBeGreaterThan(4)
    expect(undefined).not.toBeDefined()
    expect(null).toBeNull()
    expect(Number('foo')).toBeNaN()
    expect({}).toEqual({})
    expect([1]).toEqual([1])
  })
})

describe('calculator module', () => {
  // TDD STEP 1 - tests are passing, so write
  // a small test that will not pass
  describe('add function', () => {
    it('adds two numbers', () => {
      expect(add(2, 3)).toBe(5)
      expect(add(2, 2)).toBe(4)
      expect(add(2, -2)).toBe(0)
    })

    it('adds two numbers', () => {
      // stage
      const expectedResult = 5
      // act
      const actualResult = add(2, 3)
      // assert
      expect(actualResult).toBe(expectedResult)
    })

    it('adds any number of numbers')
  })
})
