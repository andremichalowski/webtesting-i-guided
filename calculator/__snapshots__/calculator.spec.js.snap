// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculator module personMaker function snapshot 1`] = `
Object {
  "age": 30,
  "foo": "bar",
  "id": "hardcodedid",
  "name": "Chris",
}
`;
